docs/articles/documentation/extending-testcafe/custom-browser-provider-plugin/index.md

```diff
@@ -0,0 +1,200 @@
+},
+
+// Optional initialization and cleanup methods
+async init () {
+    return;
+},
+
+async dispose () {
+    return;
+},
+
+// Optional methods for multi-browser support
+async getBrowserList () {
+    throw new Error('Not implemented!');
+},
+
+async isValidBrowserName (/* browserName */) {
+    return true;
+},
+
+// Optional extra methods
+async resizeWindow (/* id, width, height, currentWidth, currentHeight */) {
+    this.reportWarning('The window resize functionality is not supported by the "my-provider" browser provider.');
+},
+
+async takeScreenshot (/* id, screenshotPath, pageWidth, pageHeight */) {
+    this.reportWarning('The screenshot functionality is not supported by the "my-provider" browser provider.');
+}
+```
+
+To implement these methods, you can use the [testcafe-browser-tools](https://github.com/DevExpress/testcafe-browser-tools) library.
+
+For details, see [Browser Provider Methods](browser-provider-methods.md).
+
+### Example
+
+The following example code demonstrates how you can implement the simple browser provider for portable browsers.
+
+```js
+import path from 'path';
+import browserTools from 'testcafe-browser-tools';
+
+export default {
+    isMultiBrowser: true,
+
+    async openBrowser (id, pageUrl, browserName) {
+        var browserInfo = {};
+
+        switch (browserName) {
+            case 'firefox':
+                browserInfo.path = path.join(process.env.PORTABLE_BROWSERS_PATH, 'FirefoxPortable/FirefoxPortable.exe');
+                browserInfo.cmd  = '-new-window';
+                break;
+            case 'chrome':
+                browserInfo.path = path.join(process.env.PORTABLE_BROWSERS_PATH, 'GoogleChromePortable/GoogleChromePortable.exe');
+                browserInfo.cmd  = '--new-window';
+                break;
+            default:
+                throw new Error('Unsupported browser!');
+        }
+
+        await browserTools.open(browserInfo, pageUrl);
+    },
+
+    async closeBrowser (id) {
+        await browserTools.close(id);
+    },
+
+    async getBrowserList () {
+        return ['firefox', 'chrome'];
+    },
+
+    async isValidBrowserName (browserName) {
+        var browserList = await this.getBrowserList();
+
+        return browserList.indexOf(browserName) > -1;
+    },
+
+    async resizeWindow (id, width, height, currentWidth, currentHeight) {
+        await browserTools.resize(id, currentWidth, currentHeight, width, height);
+    },
+
+    async takeScreenshot (id, screenshotPath) {
+        await browserTools.screenshot(id, screenshotPath);
+    },
+};
+```
+
+> To know how you can implement provider methods in another way, you can explore the `src/index.js` file of the existing browser provider plugins, for example, [testcafe-browser-provider-saucelabs](https://github.com/DevExpress/testcafe-browser-provider-saucelabs/) or [testcafe-browser-provider-phantomjs](https://github.com/DevExpress/testcafe-browser-provider-phantomjs).
```

Comment:
`To learn`, not `to know`
`in a different way` not `in another way`

I'd better encourage users to `learn from actual browser providers` or something like that - not to `learn how you can do this in a different way`. That's strange.

And I don't think you need to point to ``src/index.js`` - those who read about custom browser providers should know where to look for code.